service/src/main/java/org/apache/celeborn/server/common/service/config/DbConfigServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.server.common.service.config;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.celeborn.common.CelebornConf;
+import org.apache.celeborn.server.common.service.store.IServiceManager;
+import org.apache.celeborn.server.common.service.store.db.DbServiceManagerImpl;
+
+public class DbConfigServiceImpl extends BaseConfigServiceImpl implements ConfigService {
+  private volatile IServiceManager iServiceManager;
+
+  public DbConfigServiceImpl(CelebornConf celebornConf) throws IOException {
+    super(celebornConf);
```

Comment:
Could we create JDBC connection in constuctor? When the connection could not be created, it is unnecessary to refresh cache.

Reply:
> Could we create JDBC connection in constuctor? When the connection could not be created, it is unnecessary to refresh cache.

You can refer the constructor of BaseConfigServiceImpl, when configService initialize, it will do a first refresh, if there has something wrong, server shouldn't be startup. Then it would refresh in a daemon thread.